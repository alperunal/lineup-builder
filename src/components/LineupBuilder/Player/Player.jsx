import React from 'react';
import Draggable from 'react-draggable';
import PropTypes from 'prop-types';

import goalIcon from 'assets/icons/soccer-ball.svg';
import yellowCardIcon from 'assets/icons/yellow-card.svg';
import redCardIcon from 'assets/icons/red-card.svg';
import yellowToRedCardIcon from 'assets/icons/yellow-to-red.svg';

import './Player.module.scss';

const Player = ({
    name, num, x, y, mainColor, secondaryColor, numberColor, id, goal, yellowCard, redCard, setPlayerPosition,
}) => (
    <Draggable
        bounds={{
            left: 0,
            top: 0,
            right: 320,
            bottom: 470,
        }}
        position={{ x, y }}
        onStop={(e, data) => setPlayerPosition(id, data.x, data.y)}
    >
        <div
            className="player"
            style={{
                backgroundColor: mainColor,
                border: `2px solid ${secondaryColor}`,
            }}
            id={id}
        >
            <div
                className="player__num"
                style={{
                    color: numberColor,
                }}
            >
                <span>{num}</span>
            </div>
            <div className="player__name">
                <span>{name}</span>
            </div>
            {goal && (
                <div className="player__goal">
                    <img src={goalIcon} alt="goal" />
                </div>
            )}
            {yellowCard && (
                <div className="player__card">
                    <img src={yellowCardIcon} alt="yellow card" />
                </div>
            )}
            {redCard && (
                <div className="player__card">
                    <img src={redCardIcon} alt="red card" />
                </div>
            )}
            {yellowCard && redCard && (
                <div className="player__card">
                    <img src={yellowToRedCardIcon} alt="yellow red card" />
                </div>
            )}
        </div>
    </Draggable>
);

Player.propTypes = {
    name: PropTypes.string.isRequired,
    num: PropTypes.string.isRequired,
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
    mainColor: PropTypes.string.isRequired,
    secondaryColor: PropTypes.string.isRequired,
    numberColor: PropTypes.string,
    goal: PropTypes.bool.isRequired,
    yellowCard: PropTypes.bool.isRequired,
    redCard: PropTypes.bool.isRequired,
    id: PropTypes.number.isRequired,
    setPlayerPosition: PropTypes.func.isRequired,
};

Player.defaultProps = {
    numberColor: '#FFFFFF',
};

export default Player;
