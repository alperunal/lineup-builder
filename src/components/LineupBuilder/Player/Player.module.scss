@import 'styles/_variables.scss';

.player {
  border-radius: 50%;
  cursor: pointer;
  width: 24px;
  height: 24px;
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;

  @media (min-width: $breakpoint--md) {
    width: 30px;
    height: 30px;
  }

  &__num {
    font-size: 12px;
    font-weight: bold;

    @media (min-width: $breakpoint--md) {
      font-size: 16px;
    }
  }

  &__name {
    position: absolute;
    top: 20px;
    font-size: 13px;
    font-weight: bold;
    color: #fff;
    text-shadow: 2px 2px #000;
    white-space: nowrap;

    @media (min-width: $breakpoint--md) {
      top: 25px;
      font-size: 15px;
    }
  }

  &__goal {
    position: absolute;
    left: -10px;
    top: 13px;
  }

  &__card {
    position: absolute;
    left: 15px;
    top: -13px;
  }
}
