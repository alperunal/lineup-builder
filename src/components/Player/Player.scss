.player {
    border-radius: 50%;
    cursor: pointer;
    width: 24px;
    height: 24px;
    position: absolute;
    top: 0;
    left: 0;
    display: flex;
    justify-content: center;
    align-items: center;

    @media (min-width: 720px) {
        width: 30px;
        height: 30px;
    }

    &__num {
        font-size: 12px;
        font-weight: bold;

        @media (min-width: 720px) {
            font-size: 16px;
        }
    }

    &__name {
        position: absolute;
        top: 30px;
        font-size: 13px;
        font-weight: bold;
        color: #fff;
        text-shadow: 2px 2px #000;
        white-space: nowrap;

        @media (min-width: 720px) {
            top: 35px;
            font-size: 15px;
        }
    }
}