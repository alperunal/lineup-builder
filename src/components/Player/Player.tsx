import React from "react";
import styled from "styled-components";

interface IProps {
  name: string;
  num: number;
  x: number;
  y: number;
  mainColor: string;
  secondaryColor: string;
  numberColor?: string;
}

interface IStyleProps {
  mainColor: string;
  secondaryColor: string;
  numberColor: string;
  x: number;
  y: number;
}

const Player = ({
  name,
  num,
  x,
  y,
  mainColor,
  secondaryColor,
  numberColor,
}: IProps) => {
  return (
    <Wrapper
      x={x}
      y={y}
      mainColor={mainColor}
      secondaryColor={secondaryColor}
      numberColor={numberColor}
    >
      <Num numberColor={numberColor}>
        <span>{num}</span>
      </Num>
      <Name>
        <span>{name}</span>
      </Name>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  border-radius: 50%;
  width: 24px;
  height: 24px;
  position: absolute;
  top: 0;
  left: 0;
  border: 2px solid ${({ secondaryColor }: IStyleProps) => secondaryColor};
  background-color: ${({ mainColor }: IStyleProps) => mainColor};
  transform: ${({ x, y }: IStyleProps) => `translate(${x}px, ${y}px)`};
  display: flex;
  justify-content: center;
  align-items: center;

  @media (min-width: 720px) {
    width: 36px;
    height: 36px;
    border: 3px solid ${({ secondaryColor }: IStyleProps) => secondaryColor};
  }
`;

const Num = styled.div`
  font-size: 12px;
  font-weight: bold;
  color: ${({ numberColor }: { numberColor: string }) => numberColor};

  @media (min-width: 720px) {
    font-size: 16px;
  }
`;

const Name = styled.div`
  position: absolute;
  top: 30px;
  font-size: 13px;
  font-weight: bold;
  color: #fff;
  text-shadow: 2px 2px #000;

  @media (min-width: 720px) {
    top: 42px;
    font-size: 16px;
  }
`;

export default Player;
