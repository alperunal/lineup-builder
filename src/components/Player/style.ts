import styled from 'styled-components';

interface IStyleProps {
    mainColor: string;
    secondaryColor: string;
    numberColor: string;
    x: number;
    y: number;
}

export const Wrapper = styled.div`
    border-radius: 50%;
    width: 24px;
    height: 24px;
    position: absolute;
    top: 0;
    left: 0;
    border: 2px solid ${({ secondaryColor }: IStyleProps) => secondaryColor};
    background-color: ${({ mainColor }: IStyleProps) => mainColor};
    left: ${({ x }: IStyleProps) => `${x}px`};
    top: ${({ y }: IStyleProps) => `${y}px`};
    transform: ${({ x, y }: IStyleProps) => `translate(-15px, -15px)`};
    display: flex;
    justify-content: center;
    align-items: center;

    @media (min-width: 720px) {
        width: 30px;
        height: 30px;
        border: 3px solid ${({ secondaryColor }: IStyleProps) => secondaryColor};
    }
`;

export const Num = styled.div`
    font-size: 12px;
    font-weight: bold;
    color: ${({ numberColor }: { numberColor: string }) => numberColor};

    @media (min-width: 720px) {
        font-size: 16px;
    }
`;

export const Name = styled.div`
    position: absolute;
    top: 30px;
    font-size: 13px;
    font-weight: bold;
    color: #fff;
    text-shadow: 2px 2px #000;
    white-space: nowrap;

    @media (min-width: 720px) {
        top: 35px;
        font-size: 15px;
    }
`;
