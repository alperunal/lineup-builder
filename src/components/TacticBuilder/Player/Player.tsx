import React, { useEffect } from 'react';
import Draggable from 'react-draggable';
import './Player.module.scss';

interface IProps {
    name: string;
    num: string;
    x: number;
    y: number;
    mainColor: string;
    secondaryColor: string;
    numberColor?: string;
    id: string;
    setPlayerPosition: (id: string, x: number, y: number) => void;
}

const Player: React.FC<IProps> = ({
    name, num, x, y, mainColor, secondaryColor, numberColor, id, setPlayerPosition,
}: IProps) => {
    return (
        <Draggable
            bounds={{
                left: 0,
                top: 0,
                right: 320,
                bottom: 470,
            }}
            position={{ x, y }}
            onStop={(e, data) => setPlayerPosition(id, data.x, data.y)}
        >
            <div
                className="player"
                style={{
                    backgroundColor: mainColor,
                    border: `2px solid ${secondaryColor}`,
                }}
                id={id}
            >
                <div
                    className="player__num"
                    style={{
                        color: numberColor,
                    }}
                >
                    <span>{num}</span>
                </div>
                <div className="player__name">
                    <span>{name}</span>
                </div>
            </div>
        </Draggable>
    );
}

Player.defaultProps = {
    numberColor: '#FFFFFF',
};

export default Player;
