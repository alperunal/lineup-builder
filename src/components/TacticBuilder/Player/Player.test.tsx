import React from 'react';
import { shallow } from 'enzyme';
import Player from './Player';

it('should render Player correctly', () => {
    const props = {
        name: 'Wesley Sneijder',
        x: 150,
        y: 200,
        num: '10',
        id: '10',
        mainColor: '#FFFFFF',
        secondaryColor: '#000000',
    };
    const wrapper = shallow(<Player {...props} />);
    expect(wrapper).toMatchSnapshot();
});
