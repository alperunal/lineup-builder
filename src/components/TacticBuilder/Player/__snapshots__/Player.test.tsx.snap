// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Player correctly 1`] = `ShallowWrapper {}`;
